woff1/tools/validator/validator.py

```diff
@@ -1113,16 +1110,16 @@ def _testTableDirectoryChecksums(data, reporter):
         if newChecksum != origChecksum:
             newChecksum = hex(newChecksum).strip("L")
             origChecksum = hex(origChecksum).strip("L")
-            reporter.logError(message="The \"%s\" table directory entry original checksum (%s) does not match the checksum (%s) calculated from the data." % (tag, origChecksum, newChecksum))
+            reporter.logError(message="The \"%s\" table directory entry original checksum (%s) does not match the checksum (%s) calculated from the data." % (tag.decode(), origChecksum, newChecksum))
             haveError = True
         else:
-            reporter.logPass(message="The \"%s\" table directory entry original checksum is correct." % tag)
+            reporter.logPass(message="The \"%s\" table directory entry original checksum is correct." % tag.decode())
     # check the head checksum adjustment
-    if "head" not in tables:
+    if "head".encode() not in tables.keys():
```

Comment:
String data needs to be encoded in order to be able to perform binary data comparison in Python 3.